apps/OpenSpace/ext/launcher/src/sgctedit/monitorbox.cpp

```diff
@@ -0,0 +1,216 @@
+}
+
+void MonitorBox::paintMonitorBackgrounds(QPainter& painter) {
+    painter.setPen(QPen(Qt::black, 2));
+    painter.setFont(QFont("Arial", 14));
+    for (unsigned int i = 0; i < _nMonitors; ++i) {
+        if (i <= _monitorDimensionsScaled.size()) {
+            painter.drawRect(_monitorDimensionsScaled[i]);
+            QColor fillColor("#DDDDDD");
+            QBrush brush(fillColor);
+            brush.setStyle(Qt::SolidPattern);
+            painter.fillRect(_monitorDimensionsScaled[i], brush);
+            if (_showLabel) {
+                QPointF textPos = QPointF(_monitorDimensionsScaled[i].left() + 5,
+                    _monitorDimensionsScaled[i].top() + 18);
+                painter.drawText(textPos, QString::fromStdString(std::to_string(i + 1)));
+            }
+        }
+    }
+}
+
+void MonitorBox::paintWindowBeyondBounds(QPainter& painter, const unsigned int winIdx) {
+    painter.setBrush(Qt::BDiagPattern);
+    setPenSpecificToWindow(painter, winIdx, false);
+    if (winIdx <= _windowRendering.size()) {
+        painter.drawRect(_windowRendering[winIdx]);
+    }
+    setPenSpecificToWindow(painter, winIdx, true);
+    painter.setBrush(Qt::NoBrush);
+}
+
+void MonitorBox::paintWindow(QPainter& painter, const unsigned int winIdx) {
+    setPenSpecificToWindow(painter, winIdx, true);
+    if (winIdx <= _windowRendering.size()) {
+        painter.drawRect(_windowRendering[winIdx]);
+        QColor fillColor = QColor(_colorsForWindows[winIdx]);
+        fillColor.setAlpha(_alphaWindowOpacity);
+        QBrush brush(fillColor);
+        brush.setStyle(Qt::SolidPattern);
+        painter.fillRect(_windowRendering[winIdx], brush);
+        paintWindowNumber(painter, winIdx);
+    }
+}
+
+void MonitorBox::paintWindowNumber(QPainter& painter, const unsigned int winIdx) {
+    QPointF textPos = QPointF(_windowRendering[winIdx].left() + 5,
+        _windowRendering[winIdx].bottom() - 5);
+    textPos.setX(std::clamp(textPos.x(), 0.0, _monitorWidgetSize.width() - 10));
+    textPos.setY(std::clamp(textPos.y(), 0.0, _monitorWidgetSize.height() - 10));
+    painter.drawText(textPos, QString::fromStdString(std::to_string(winIdx + 1)));
+}
+
+void MonitorBox::setPenSpecificToWindow(QPainter& painter, const unsigned int windowIdx,
+                                                                       bool visibleBorder)
+{
+    int penWidth = (visibleBorder) ? 1 : -1;
+    painter.setPen(QPen(QColor(_colorsForWindows[windowIdx]), penWidth));
+}
+
+void MonitorBox::windowDimensionsChanged(unsigned int monitorIdx, unsigned int windowIdx,
+                                                              const QRectF& newDimensions)
+{
+    mapWindowResolutionToWidgetCoordinates(monitorIdx, windowIdx, newDimensions);
+}
+
+void MonitorBox::mapMonitorResolutionToWidgetCoordinates() {
+    QSize virtualDesktopResolution;
+    float maxWidth = 0.0;
+    float maxHeight = 0.0;
+    for (auto m : _monitorResolution) {
```

Comment:
-> `const QRect&` instead of `auto`

Reply:
I made this change. Unlike my previous defenses for the use of `auto`, the scope here is not as limited.